server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,138 +290,348 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
+      LOG.info("Not updating lookups because process was interrupted or not finished starting yet.");
+      return;
+    }
+
+    final Map<String, Map<String, LookupExtractorFactoryMapContainer>> allLookupTiers = lookupMapConfigRef.get();
+
+    if (allLookupTiers == null) {
+      LOG.info("Not updating lookups because no data exists");
+      return;
+    }
+
+    LOG.debug("Starting lookup sync for on all nodes.");
+
+    final Map<HostAndPort, LookupsState<LookupExtractorFactoryMapContainer>> currState = new ConcurrentHashMap<>();
+
+    try {
+      List<ListenableFuture<Map.Entry>> futures = new ArrayList<>();
+      for (Map.Entry<String, Map<String, LookupExtractorFactoryMapContainer>> tierEntry : allLookupTiers.entrySet()) {
+
+        LOG.debug("Starting lookup mgmt for tier [%s].", tierEntry.getKey());
+
+        final Map<String, LookupExtractorFactoryMapContainer> tierLookups = tierEntry.getValue();
+        for (final HostAndPort node : listenerDiscoverer.getNodes(LookupModule.getTierListenerPath(tierEntry.getKey()))) {
+
+          LOG.debug(
+              "Starting lookup mgmt for tier [%s] and host [%s:%s].",
+              tierEntry.getKey(),
+              node.getHostText(),
+              node.getPort()
+          );
+
+          futures.add(
+              executorService.submit(
+                  () -> {
+                    try {
+                      return new AbstractMap.SimpleImmutableEntry<>(node, doLookupManagementOnNode(node, tierLookups));
+                    }
+                    catch (Exception ex) {
+                      LOG.makeAlert(
+                          ex,
+                          "Failed to finish lookup management on node [%s:%s]",
+                          node.getHostText(),
+                          node.getPort()
+                      );
+                      return null;
+                    }
+                  }
+              )
+          );
         }
-      });
-      started = true;
-      LOG.debug("Started");
+      }
+
+      final ListenableFuture<List<Map.Entry>> allFuture = Futures.allAsList(futures);
+      try {
+        ImmutableMap.Builder<HostAndPort, LookupsState<LookupExtractorFactoryMapContainer>> stateBuilder = ImmutableMap.builder();
+        allFuture.get(lookupCoordinatorManagerConfig.getAllHostTimeout().getMillis(), TimeUnit.MILLISECONDS)
+                 .stream()
+                 .filter(Objects::nonNull)
+                 .forEach(stateBuilder::put)
+        ;
+        knownOldState.set(stateBuilder.build());
+      }
+      catch (InterruptedException ex) {
+        allFuture.cancel(true);
+        Thread.currentThread().interrupt();
+        throw ex;
+      } catch (Exception ex) {
+        allFuture.cancel(true);
+        throw ex;
+      }
+
+    } catch (Exception ex) {
+      LOG.makeAlert(ex, "Failed to finish lookup management loop.").emit();
```

Comment:
If `Thread.currentThread().interrupt();` was called to get here, will it even successfully emit an alert?